appsec-kit-v8/src/main/java/com/vaadin/appsec/v8/ui/content/VulnerabilityDetailsView.java

```diff
@@ -218,21 +228,72 @@ private Component buildVaadinAnalysis() {
         vaTitle.setWidth(100, Unit.PERCENTAGE);
         vAnalysisPanel.addComponent(vaTitle);
 
-        // TODO Replace with real data when available
-        Label vaStatus = new Label("Under review");
-        vaStatus.addStyleName("va-under-review");
+        boolean isSupportedVaadinVersion = false;
+        String vaadinVersion = Version.getFullVersion();
+        AppSecService appsec = AppSecService.getInstance();
+        List<String> fw7Versions = appsec.getSupportedFramework7Versions();
+        List<String> fw8Versions = appsec.getSupportedFramework8Versions();
+        if (fw7Versions.contains(vaadinVersion)
+                || fw8Versions.contains(vaadinVersion)) {
+            isSupportedVaadinVersion = true;
+        }
+
+        Label vaStatus = new Label();
+        Label vaDesc = new Label();
+        if (!isSupportedVaadinVersion) {
+            vaStatus.setValue("Not Provided");
+            vaStatus.addStyleName("va-not-supported");
+            vaDesc.setValue("This app is running with Vaadin " + vaadinVersion
+                    + " which is not currently getting security updates. Please "
+                    + " upgrade to one of the latest maintained versions: "
+                    + (vaadinVersion.startsWith("7")
+                            ? fw7Versions.stream()
```

Comment:
Same question about the `String.join()` here.

Reply:
Same answer: yes 🙂